tvb_contrib/tvb/contrib/scripts/utils/data_structures_utils.py

```diff
@@ -868,24 +871,28 @@ def data_xarray_from_continuous_events(events, times, senders, variables=[],
         from xarray import DataArray
         return DataArray(data, dims=list(coords.keys()), coords=coords, name=name)
     except:
+        # Return a dictionary as a plan B'
         return {"data": data, "dims": list(coords.keys()), "coords": coords, "name": name}
 
 
-def concatenate_heterogeneous_DataArrays(data, dim_name, data_name="", fill_value=np.nan, dims=None):
+def concatenate_heterogeneous_DataArrays(data, concat_dim_name,
+                                         data_keys=None, name=None, fill_value=np.nan, transpose_dims=None):
+    from pandas import Series
```

Comment:
ok more libs here, neither xarray nor pandas are on the dependency list for TVB.  since it's in contrib, we can leave it but consider how to handle this in production, as an optional dependency.